pkg/kotsutil/kots.go

```diff
@@ -671,8 +671,9 @@ func GetInstallationParams(configMapName string) (InstallationParams, error) {
 		return autoConfig, errors.Wrap(err, "failed to get k8s clientset")
 	}
 
-	kotsadmConfigMap, err := clientset.CoreV1().ConfigMaps(util.PodNamespace).Get(context.TODO(), configMapName, metav1.GetOptions{})
+	autoConfig.EnableImageDeletion = IsKurl(clientset)
 
+	kotsadmConfigMap, err := clientset.CoreV1().ConfigMaps(util.PodNamespace).Get(context.TODO(), configMapName, metav1.GetOptions{})
```

Comment:
this configmap doesn't exist in kurl (and there was a reason for that that i can't remember.), so `EnableImageDeletion` is always set to false and garbage collection didn't happen.